src/sources/datadog_agent/mod.rs

```diff
@@ -334,7 +334,8 @@ impl DatadogAgentSource {
                     .expect("static regex always compiles"),
             },
             log_schema_host_key: log_schema().host_key(),
-            log_schema_source_type_key: log_schema().source_type_key(),
+            log_schema_source_type_key: log_schema().source_type_key()
+                .map_or("", |key| Box::leak(key.to_string().into_boxed_str())),
```

Comment:
An alternative here is to change `log_schema_source_type_key` which is `&'static str` to `String`. That would also require changing a couple of [places](https://github.com/search?q=repo%3Avectordotdev%2Fvector%20%22source.log_schema_source_type_key%2C%22&type=code) to copy.

Reply:
I would prefer to change this to a `String`. This is run at boot time so we shouldn't need to worry about an extra allocation and it just makes for nicer code..

Reply:
Makes sense. Changed to `String`.